cmake/Macros.cmake

```diff
@@ -301,11 +301,39 @@ macro(sfml_add_example target)
 
 endmacro()
 
+# add a new static library which provides a test runner with SFML testing utilities
+# example: sfml_add_test_runner(sfml-test-runner-window
+#                               SOURCES TestUtilities/WindowUtil.hpp TestUtilities/WindowUtil.cpp
+#                               DEPENDS SFML::Window sfml-test-runner-system)
+function(sfml_add_test_runner target)
+
+    # parse the arguments
+    cmake_parse_arguments(THIS "" "" "SOURCES;DEPENDS" ${ARGN})
+    if (NOT "${THIS_UNPARSED_ARGUMENTS}" STREQUAL "")
+        message(FATAL_ERROR "Extra unparsed arguments when calling sfml_add_test_runner: ${THIS_UNPARSED_ARGUMENTS}")
+    endif()
+
+    # set a source group for the source files
+    source_group("" FILES ${THIS_SOURCES})
+
+    # create the target
+    add_library(${target} STATIC ${THIS_SOURCES})
+
+    # set the target's folder (for IDEs that support it, e.g. Visual Studio)
+    set_target_properties(${target} PROPERTIES FOLDER "Tests/Runners")
+
+    # link the target to its SFML dependencies and its test runner library
+    target_link_libraries(${target} PUBLIC ${THIS_DEPENDS})
+
+    set_target_warnings(${target})
+endfunction()
+
```

Comment:
Can we move this to the test/ directory since nothing outside of the unit tests needs to use this? For starters we can put this in test/CMakeLists.txt.

Reply:
Yes, that seems reasonable.

Reply:
Done